### PR TITLE
chore: use whitelist for alby events

### PR DESCRIPTION
**Pull request by rolznz**
Fixes https://github.com/getAlby/hub/issues/530

### CODE REVIEW

File: alby/alby_oauth_service.go [hunk 1]
```diff
@@ -645,11 +646,38 @@ func (svc *albyOAuthService) ConsumeEvent(ctx context.Context, event *events.Eve
...
+
+	if !slices.Contains(eventWhitelist, event.Event) {
+		logger.Logger.WithField("event", event).Debug("Skipped sending non-whitelisted event to alby events API")
+		return
+	}
 
 	// TODO: rename this config option to be specific to the alby API
 	if !svc.cfg.GetEnv().LogEvents {
-		logger.Logger.WithField("event", event).Debug("Skipped sending to alby events API")
+		logger.Logger.WithField("event", event).Debug("Skipped sending to alby events API (alby event logging disabled)")
```
Comment (bumi) [inline alby/alby_oauth_service.go:680]: nit: first check if it's enabled and if it's enabled we do the whitelist check?

File: alby/alby_oauth_service.go [hunk 2]
```diff
@@ -645,11 +646,38 @@ func (svc *albyOAuthService) ConsumeEvent(ctx context.Context, event *events.Eve
 		return
 	}
 
-	// TODO: we should have a whitelist rather than a blacklist, so new events are not automatically sent
+	// ensure we do not send unintended events to Alby API
+	eventWhitelist := []string{
```
Comment (bumi) [inline alby/alby_oauth_service.go:650]: can we move this outside of the function into some constant?

File: alby/alby_oauth_service.go (hunk 1 above)
Comment (rolznz) [reply alby/alby_oauth_service.go:680]: It's just a bit harder to test without enabling events in dev mode. But I will switch it

File: alby/alby_oauth_service.go (hunk 2 above)
Comment (rolznz) [reply alby/alby_oauth_service.go]: I moved it to a separate function (unfortunately in go there is no constant for arrays)

**Review by im-adithya — APPROVED** [review]
utACK

**Pull request merged**